cogito/put_test.go

```diff
@@ -212,20 +212,79 @@ func TestPutterLoadConfigurationInvalidParamsFailure(t *testing.T) {
 	assert.Error(t, err, wantErr)
 }
 
+func TestPutterLoadConfigurationInvalidSinksFailure(t *testing.T) {
+	in := []byte(`
+	{
+	  "source": {"sinks": ["gchat"], "gchat_webhook": "sensitive-webhook"},
+	  "params": {"sinks": ["margherita"]}
+	}`)
+	wantErr := "put: arguments: unsupported sink: margherita"
+	putter := cogito.NewPutter("dummy-API", hclog.NewNullLogger())
+
+	err := putter.LoadConfiguration(in, nil)
+
+	assert.Error(t, err, wantErr)
+}
+
+func TestPutterLoadConfigurationMissingGchatwebHook(t *testing.T) {
+	in := []byte(`
+{
+  "source": {"sinks": ["gchat"]},
+  "params": {}
+}`)
+	wantErr := `put: source: missing keys: gchat_webhook`
+	putter := cogito.NewPutter("dummy-API", hclog.NewNullLogger())
+
+	err := putter.LoadConfiguration(in, nil)
+
+	assert.Error(t, err, wantErr)
+}
+
+func TestPutterLoadConfigurationUnknownSink(t *testing.T) {
+	in := []byte(`
+{
+  "source": {"sinks": ["pizza"]},
+  "params": {}
+}`)
+	wantErr := `put: source: invalid sink(s): [pizza]. Supported sinks: [gchat github]`
+	putter := cogito.NewPutter("dummy-API", hclog.NewNullLogger())
+
+	err := putter.LoadConfiguration(in, nil)
+
+	assert.Error(t, err, wantErr)
+}
+
+func TestPutterLoadConfigurationUnknownSinkPutParams(t *testing.T) {
```

Comment:
What is the difference with? `TestPutterLoadConfigurationInvalidSinksFailure`?

Same error message, only difference is in sinks name. Either I am missing something or this is the same test.

Reply:
Good catch, thanks. You are right they are testing the same.